### PR TITLE
feat(ExtendedSplashScreen): add android 12 splashscreen support

### LINKED ISSUE #1054: [ExtendedSplashScreen][Android] Content does not match native splash screen
> I'm trying to use the Uno Toolkit's ExtendedSplashScreen control to create an extended splash screen for our app using this
[documentation](https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/ExtendedSplashScreen.html). While I'm now able to display an extended splash screen, it's slightly different from the native splashscreen, causing a jarring transition. I was wondering if you had any guidance for how to make sure the extended splashscreen can match exactly the native splash screen (plus a loading wheel of some kind). This functionality is why we previously used nventive's
[ExtendedSplashScreen library](https://github.com/nventive/ExtendedSplashScreen).
_- @juliecantin on Teams_

> The video was taken on an emulator running android 11. The sample was created by the Uno wizard, so I imagine it targets the latest version of the SDK. The app I'm having issues with, targets Uno 4 and Android SDK 13

[extendedsplash.webm](https://github.com/unoplatform/uno.toolkit.ui/assets/4793020/1cd2f3ac-4935-4436-aa57-40ca26f892fb)

> I also tested the extended splash screen on my physical pixel 6 Android 13 device, and here the extended splash screen doesn't seem to automatically include the native splash screen at all. Recording here (from the same sample app, no change except to make the background of the main page white)

https://github.com/unoplatform/uno.toolkit.ui/assets/4793020/15ad93f6-4f0b-4b24-bd0f-e0e080a5fb4a

Repro app:
[extendedsplash.zip](https://github.com/unoplatform/uno.toolkit.ui/files/14435730/extendedsplash.zip)

### PR DESCRIPTION
**Pull request by kazo0**
fixes #1054 

Left is API 34, Right is API 28
splash-ezgif com-video-to-gif-converter

### CODE REVIEW

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Mobile/Android/MainActivity.Android.cs
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Mobile/Android/MainActivity.Android.cs:1]: no needed anymore?

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml [hunk 1]
```diff
@@ -11,64 +11,98 @@
...
-			<muxc:NavigationView.PaneHeader>
-				<!-- Left padding for overlay toggle button -->
-				<Grid Padding="24,8,0,0">
-					<!-- Image -->
-					<Image Source="ms-appx:///Assets/UnoLogo.png"
-						   Height="101"
-						   Width="75"
-						   HorizontalAlignment="Left"
-						   Margin="16,8,0,12" />
+					<!--<Image Source="ms-appx:///SplashScreenApp/Assets/Icons/splash.png"
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml:28]: not keeping hte image?

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml [hunk 2]
```diff
@@ -11,64 +11,98 @@
...
-		<Grid.RowDefinitions>
-			<RowDefinition Height="Auto" />
-			<RowDefinition Height="*" />
-		</Grid.RowDefinitions>
+	<utu:ExtendedSplashScreen x:Name="Splash"
+							  HorizontalAlignment="Stretch"
+							  VerticalAlignment="Stretch"
+							  HorizontalContentAlignment="Stretch"
+							  VerticalContentAlignment="Stretch">
+		<utu:ExtendedSplashScreen.LoadingContentTemplate>
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml:19]: can probably move data-template to a resource

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml [hunk 3]
```diff
@@ -11,64 +11,98 @@
...
-					<Image Source="ms-appx:///Assets/UnoLogo.png"
-						   Height="101"
-						   Width="75"
-						   HorizontalAlignment="Left"
-						   Margin="16,8,0,12" />
+					<!--<Image Source="ms-appx:///SplashScreenApp/Assets/Icons/splash.png"
+						   Grid.Row="1"
+						   VerticalAlignment="Center"
+						   Width="192"
+						   Height="192" />-->
```
Comment (pictos) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml:32]: do we need this?

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml:1]: format the file to adjust tab level

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs [hunk 4]
```diff
@@ -277,4 +280,37 @@ private void NavigationViewControl_DisplayModeChanged(MUXC.NavigationView sender
 			}
 		}
 	}
+
+	public class MyLoadableSource : ILoadable
+	{
+
+		public event EventHandler? IsExecutingChanged;
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs:287]:
```suggestion
		public event EventHandler IsExecutingChanged;
```
so we dont keep getting this github actions check on subsequent prs

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Mobile/Android/MainActivity.Android.cs
Comment (Xiaoy312) [reply samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Mobile/Android/MainActivity.Android.cs:1]: same for winui one

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs [hunk 5]
```diff
@@ -277,4 +280,37 @@ private void NavigationViewControl_DisplayModeChanged(MUXC.NavigationView sender
...
+				}
+			}
+		}
+
+		public async Task Execute()
+		{
+			try
+			{
+				IsExecuting = true;
+				await Task.Delay(5000);
```
Comment (Xiaoy312) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs:308]: maybe add a comment saying this delay is intentional?

File: src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs [hunk 6]
```diff
@@ -10,55 +10,269 @@
...
+using AndroidSplashScreen = AndroidX.Core.SplashScreen.SplashScreen;
+using Uno.Toolkit.UI.Controls;
+using Android;
+using Windows.Devices.Sensors;
+using Android.Content.Res;
+
+
+
+
 #if IS_WINUI
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs:31]:
```suggestion
using Android.Content.Res;

#if IS_WINUI
```

**Review by pictos — APPROVED** [review]
LGTM! Just some comments that I think we can delete, but it doesn't invalidate the PR

File: src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs [hunk 7]
```diff
@@ -10,55 +10,269 @@
...
+		}
+
+		if (Instance is not null)
+		{
+			Instance.SplashScreenContent = await GetNativeSplashScreen(null) ?? new Border();
+		}
+
+		
+
+		// Remove the native splash now that we've set the extended one.
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs:142]:
```suggestion
		}

		// Remove the native splash now that we've set the extended one.
```

File: src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs [hunk 8]
```diff
@@ -10,55 +10,269 @@
...
+			};
+		}
+	}
+
+	private static Bitmap? GetBitmapFromView(View view)
+	{
+		var returnedBitmap = Bitmap.CreateBitmap(view.Width, view.Height, Bitmap.Config.Argb8888!);
+		if (returnedBitmap is not null)
+		{
+			Android.Graphics.Canvas canvas = new Android.Graphics.Canvas(returnedBitmap);
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Android.cs:209]:
```suggestion
			var canvas = new Android.Graphics.Canvas(returnedBitmap);
```

File: src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/SystemUiConfig.Android.cs
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/SystemUiConfig.Android.cs:1]: could just inline this into ExtendedSplashScreen.Android.cs as a record (drop the ? on `OriginalSystemUi` if using `record`, or keep for `record struct`)

**Review by Xiaoy312 — APPROVED** [review]

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs (hunk 5 above)
Comment (kazo0) [reply samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/Shell.xaml.cs:308]: ah, i missed a bunch of debugging stuff that shouldn't be here, shouldn't be using this

**Review by agneszitte — APPROVED** [review]

**Pull request merged**